src/contrib/persistence/Akka.Persistence.Sqlite.Tests/CustomObjectSerializerSpec.cs

```diff
@@ -0,0 +1,194 @@
+using Microsoft.Data.Sqlite;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Persistence.Sqlite.Tests
+{
+    public class CustomObjectSerializerSpec : Akka.TestKit.Xunit2.TestKit, IAsyncLifetime
+    {
+        private static readonly string ConnectionString;
+        private static readonly Config Config;
+        static CustomObjectSerializerSpec()
+        {
+            var filename = $"AkkaSqlite-{Guid.NewGuid()}.db";
+            File.Copy("./data/Sqlite.CustomObject.db", $"{filename}.db");
+            
+            ConnectionString = $"DataSource={filename}.db";
+            Config = ConfigurationFactory.ParseString($@"
+akka.actor {{
+    serializers {{
+        mySerializer = ""{typeof(MySerializer).AssemblyQualifiedName}""
+    }}
+    serialization-bindings {{
+        ""System.Object"" = mySerializer
+    }}
+}}
+
+akka.persistence {{
+    journal {{
+        plugin = ""akka.persistence.journal.sqlite""
+        sqlite {{
+            connection-string = ""{ConnectionString}""
+            auto-initialize = on
+        }}
+    }}
+    snapshot-store {{
+        plugin = ""akka.persistence.snapshot-store.sqlite""
+        sqlite {{
+            connection-string = ""{ConnectionString}""
+            auto-initialize = on
+        }}
+    }}
+}}").WithFallback(SqlitePersistence.DefaultConfiguration());
+        }
+
+        public CustomObjectSerializerSpec(ITestOutputHelper helper) 
+            : base(Config, nameof(CustomObjectSerializerSpec), helper)
+        {
+        }
+
+        [Fact(DisplayName = "Persistence.Sql should use custom serializer for object type")]
+        public async Task CustomSerializerTest()
+        {
+            var probe = CreateTestProbe();
+            
+            // Sanity check to see that the system should serialize object type using MySerializer
+            var serializer = Sys.Serialization.FindSerializerForType(typeof(Persisted));
+            serializer.Should().BeOfType<MySerializer>();
+            
+            var actor = Sys.ActorOf(Props.Create(() => new PersistedActor("a", probe)));
+            probe.ExpectMsg("recovered");
+            actor.Tell(new Persisted("a"), probe);
+            probe.ExpectMsg(new Persisted("a"));
+
+            // Read the database directly, make sure that we're using the correct object type serializer
+            var conn = new SqliteConnection(ConnectionString);
+            conn.Open();
+            const string sql = "SELECT ej.serializer_id FROM event_journal ej WHERE ej.persistence_id = 'a'";
+            await using var cmd = new SqliteCommand(sql, conn);
+            var record = await cmd.ExecuteReaderAsync();
+            await record.ReadAsync();
+            
+            // In the bug this fails, the serializer id is JSON id instead of MySerializer id
+            record[0].Should().Be(9999);
+        }
+
+        [Fact(DisplayName = "Persistence.Sql should be able to read legacy data")]
+        public void LegacyDataTest()
```

Comment:
So to make this cover the issue referenced here: https://github.com/akkadotnet/akka.net/pull/6223/files#r1012961321 - we just need even older data, correct?